components/kyma-environment-broker/internal/process/deprovisioning/event_hub_step.go

```diff
@@ -0,0 +1,129 @@
+package deprovisioning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	processazure "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+)
+
+type DeprovisionAzureEventHubStep struct {
+	OperationManager *process.DeprovisionOperationManager
+	instanceStorage  storage.Instances
+	processazure.EventHub
+}
+
+func NewDeprovisionAzureEventHubStep(os storage.Operations, is storage.Instances, hyperscalerProvider azure.HyperscalerProvider, accountProvider hyperscaler.AccountProvider, ctx context.Context) DeprovisionAzureEventHubStep {
+	return DeprovisionAzureEventHubStep{
+		OperationManager: process.NewDeprovisionOperationManager(os),
+		instanceStorage:  is,
+		EventHub: processazure.EventHub{
+			HyperscalerProvider: hyperscalerProvider,
+			AccountProvider:     accountProvider,
+			Context:             ctx,
+		},
+	}
+}
+
+var _ Step = (*DeprovisionAzureEventHubStep)(nil)
+
+func (s DeprovisionAzureEventHubStep) Name() string {
+	return "Deprovision Azure Event Hubs"
+}
+
+func (s DeprovisionAzureEventHubStep) Run(operation internal.DeprovisioningOperation, log logrus.FieldLogger) (internal.DeprovisioningOperation, time.Duration, error) {
```

Comment:
The step should be written the way it can be run many times even if it succeeded after the first time. All necessary data like IDs can be set in the DeprovisioningOperation in the initialization step. Look at AvS deprovisioning as an example. Currently every time the step is executed, it gets instance from the storage and interacts with Azure even if everything was deprovisioned minute ago. You can even save in the deprovisioning operation the ProvisioningParameters and then use it from deprovisioningoperation instance. If the comment is not clear, please contact with me

Reply:
please, read the doc: docs/kyma-environment-broker/03-03-runtime-provisioning-and-deprovisioning.md

Reply:
@piotrmiskiewicz Thanks for the review.
According to my understanding to your comment. You want to save the work done if the Event Hub was already deprovisioned before. right?
In the beginning we query the instance, if it was deleted/deprovisioned we return/succeed the operation. Is this enough?

Reply:
@piotrmiskiewicz 
We have a couple of points here:
1- We don’t maintain the Event Hub life cycle in the Provision step, so one idea is to implement the maintaining the state of the provision/deprovision in a sperate ticket.
2- The deprovision step is now idempotent as long as we check for the instance existence in the beginning and IMHO this is better than maintaining the state of the operation in a DB.
What do you think?

Reply:
According to my understanding to your comment. You want to save the work done if the Event Hub was already deprovisioned before. right? 
Yes

In the beginning we query the instance, if it was deleted/deprovisioned we return/succeed the operation. Is this enough?
No, the instance can exists but event hub is deprovisioned because the process of deprovisonig is ongoing (other systems are being deprovisioned)

Reply:
Maintaining the state of deprovisioning/provisioning is easy - see how it is done in AvS step.

Reply:
The provisioning process of event hub does not maintain the state and every time it takes long time even if the work is unnecessary, which is not nice.

Reply:
@piotrmiskiewicz Can you please check now?
Now we persist in the DB the deletion status.